src/AspNet.Security.OAuth.Zoho/ZohoAuthenticationPostConfigureOptions.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.Zoho;
+
+/// <summary>
+/// Used to configure <see cref="ZohoAuthenticationOptions"/> instances.
+/// </summary>
+public sealed class ZohoAuthenticationPostConfigureOptions : IPostConfigureOptions<ZohoAuthenticationOptions>
+{
+    /// <inheritdoc />
+    public void PostConfigure(
+        string? name,
+        [NotNull] ZohoAuthenticationOptions options)
+    {
+        ConfigureEndpoints(options);
+    }
+
+    private static void ConfigureEndpoints(ZohoAuthenticationOptions options)
+    {
+        if (AreEndpointsInitialized(options))
+        {
+            return;
+        }
```

Comment:
Test each one on an individual basis, rather than as a group.

Reply:
What do you think if I add one more Region 'Custom'? And use it like:

```
        if (options.Region == ZohoAuthenticationRegion.Custom)
        {
            return;
        }

        var domain = GetDomain(options.Region);

        options.AuthorizationEndpoint = CreateUrl(domain, ZohoAuthenticationDefaults.AuthorizationPath);
        options.TokenEndpoint = CreateUrl(domain, ZohoAuthenticationDefaults.TokenPath);
        options.UserInformationEndpoint = CreateUrl(domain, ZohoAuthenticationDefaults.UserInformationPath);
```

then the user can use any future regions/endpoints.

Reply:
I would just guard the property assignments with a null check and not set them if they already have a value.